Security/Authentication/Provider/UserAuthenticationProvider.php

```
@@ -149,16 +149,15 @@ private function authenticateUser(TokenInterface $token, UserInterface $user)
      */
     private function authenticateWithEncoder(TokenInterface $token, UserInterface $user)
     {
-        try {
-            $classname = \Symfony\Component\Security\Core\Util\ClassUtils::getRealClass($user);
-            if (true === $this->_encoderFactory
-                            ->getEncoder($classname)
-                            ->isPasswordValid($user->getPassword(), $token->getCredentials(), $user->getSalt())) {
-                return new UsernamePasswordToken($user, $user->getPassword(), $user->getRoles());
-            }
-        } catch (Exception $e) {
-            return false;
+        $classname = \Symfony\Component\Security\Core\Util\ClassUtils::getRealClass($user);
+
+        if (true === $this->_encoderFactory
+                ->getEncoder($classname)
+                ->isPasswordValid($user->getPassword(), $token->getCredentials(), $user->getSalt())) {
+            return new UsernamePasswordToken($user, $user->getPassword(), $user->getRoles());
         }
```

Hey @eric-chau. I can do it but in the catch, you want to throw the exception or just return false ?